tests/models/pop2piano/test_feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,218 @@
+        speech_input = np.zeros(
+            [
+                1000000,
+            ],
+            dtype=np.float32,
+        )
+
+        input_features = feature_extractor(speech_input, audio_sr=16_000, return_tensors="np")
+        self.assertTrue(input_features.input_features.ndim == 3)
+        self.assertEqual(input_features.input_features.shape[-1], 512)
+
+        self.assertTrue(input_features.beatsteps.ndim == 1)
+        self.assertTrue(input_features.ext_beatstep.ndim == 1)
+
+    def _load_datasamples(self, num_samples):
+        ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
+        # automatic decoding with librispeech
+        speech_samples = ds.sort("id").select(range(num_samples))[:num_samples]["audio"]
+
+        return [x["array"] for x in speech_samples], [x["sampling_rate"] for x in speech_samples]
+
+    def test_integration(self):
+        EXPECTED_INPUT_FEATURES = torch.tensor(
+            [[-7.1493, -6.8701, -4.3214], [-5.9473, -5.7548, -3.8438], [-6.1324, -5.9018, -4.3778]]
+        )
+
+        input_speech, sampling_rate = self._load_datasamples(1)
+        feaure_extractor = Pop2PianoFeatureExtractor.from_pretrained("susnato/pop2piano_dev")
+        input_features = feaure_extractor(input_speech, audio_sr=sampling_rate[0], return_tensors="pt").input_features
+        self.assertTrue(torch.allclose(input_features[0, :3, :3], EXPECTED_INPUT_FEATURES, atol=1e-4))
+
+    @unittest.skip("Pop2PianoFeatureExtractor does not return attention_mask")
+    def test_attention_mask(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor does not return attention_mask")
+    def test_attention_mask_with_truncation(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor only takes one raw_audio at a time")
+    def test_batch_feature_pt(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor only takes one raw_audio at a time")
+    def test_batch_feature_tf(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor only takes one raw_audio at a time")
+    def test_batch_feature(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor does not supports padding")
+    def test_padding_accepts_tensors_pt(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor does not supports padding")
+    def test_padding_accepts_tensors_tf(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor does not supports truncation")
+    def test_truncation_from_array(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor does not supports truncation")
+    def test_truncation_from_list(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor does not supports padding")
+    def test_padding_from_list(self):
+        pass
+
+    @unittest.skip("Pop2PianoFeatureExtractor does not supports padding")
+    def test_padding_from_array(self):
+        pass
```

Comment:
Hi @ArthurZucker If I am not wrong, I think we already talked about this [here](https://github.com/huggingface/transformers/pull/21785#discussion_r1142218564) as it would complicate things since all the audio files needed to be of same length. As @sweetcocoa pointed out in the original training - 

> In fact, for training, batch audio inputs of fixed length and corresponding pre-computed features were used (without using feature extractor on-the-fly).